Source/CesiumRuntime/Private/CesiumEncodedMetadataUtility.cpp

```diff
@@ -71,232 +75,239 @@ EncodedPixelFormat getPixelFormat(
+      gpuType = ECesiumMetadataPackedGpuType_DEPRECATED::Float_DEPRECATED;
+    }
+
+    if (pExpectedProperty->Normalized != isNormalized) {
+      if (isNormalized) {
+        UE_LOG(
+            LogCesium,
+            Warning,
+            TEXT("Unexpected normalization in feature table property."));
+      } else {
+        UE_LOG(
+            LogCesium,
+            Warning,
+            TEXT("Feature table property not normalized as expected"));
+      }
+      continue;
+    }
+
+    // Only support normalization of uint8 for now
+    if (isNormalized &&
+        trueType.ComponentType != ECesiumMetadataComponentType::Uint8) {
+      UE_LOG(
+          LogCesium,
+          Warning,
+          TEXT(
+              "Feature table property has unexpected type for normalization, only normalization of Uint8 is supported."));
+      continue;
+    }
+
+    EncodedPixelFormat encodedFormat =
+        getPixelFormat(gpuType, componentCount, isNormalized);
+
+    if (encodedFormat.format == EPixelFormat::PF_Unknown) {
+      UE_LOG(
+          LogCesium,
+          Warning,
+          TEXT(
+              "Unable to determine a suitable GPU format for this feature table property."));
+      continue;
+    }
+
+    TRACE_CPUPROFILER_EVENT_SCOPE(Cesium::EncodePropertyArray)
+
+    EncodedMetadataProperty& encodedProperty =
+        encodedFeatureTable.encodedProperties.Emplace_GetRef();
+    encodedProperty.name =
+        "FTB_" + featureTableDescription.Name + "_" + pair.Key;
+
+    int64 floorSqrtFeatureCount = glm::sqrt(featureCount);
+    int64 ceilSqrtFeatureCount =
+        (floorSqrtFeatureCount * floorSqrtFeatureCount == featureCount)
+            ? floorSqrtFeatureCount
+            : (floorSqrtFeatureCount + 1);
+    encodedProperty.pTexture = MakeUnique<LoadedTextureResult>();
+    // TODO: upgrade to new texture creation path.
+    encodedProperty.pTexture->textureSource = LegacyTextureSource{};
+    encodedProperty.pTexture->pTextureData = createTexturePlatformData(
+        ceilSqrtFeatureCount,
+        ceilSqrtFeatureCount,
+        encodedFormat.format);
+
+    encodedProperty.pTexture->addressX = TextureAddress::TA_Clamp;
+    encodedProperty.pTexture->addressY = TextureAddress::TA_Clamp;
+    encodedProperty.pTexture->filter = TextureFilter::TF_Nearest;
+
+    if (!encodedProperty.pTexture->pTextureData) {
+      UE_LOG(
+          LogCesium,
+          Error,
+          TEXT(
+              "Error encoding a feature table property. Most likely could not allocate enough texture memory."));
+      continue;
+    }
+
+    FTexture2DMipMap* pMip = new FTexture2DMipMap();
+    encodedProperty.pTexture->pTextureData->Mips.Add(pMip);
+    pMip->SizeX = ceilSqrtFeatureCount;
+    pMip->SizeY = ceilSqrtFeatureCount;
+    pMip->BulkData.Lock(LOCK_READ_WRITE);
+
+    void* pTextureData = pMip->BulkData.Realloc(
+        ceilSqrtFeatureCount * ceilSqrtFeatureCount * encodedFormat.pixelSize);
+
+    if (isArray) {
```

Comment:
Correct me if I'm wrong, but I don't think this code ever accounted for normalization. So if there was a normalized uint property 0-255, it would just get encoded as floats 0.0f-255.0f, and it wouldn't actually denormalize the values.

I think that's the case, and if so, I'll add the correct handling of this in a follow-up PR